app/decorators/nav/link_builder.rb

```diff
@@ -0,0 +1,280 @@
+# frozen_string_literal: true
+
+# Defines nav menus and related helper methods.
+module Nav
+  class LinkBuilder < ApplicationDecorator
+
+    delegate_all
+
+    attr_accessor :context
+
+    def initialize()
+      context = {}
```

Comment:
Ruby quirk: you have to do `self.context = {}`. Otherwise you're setting a local variable. If you install the rubocop linter plugin for Atom it will help you catch these things.

Reply:
Thanks for the heads-up!  I will run the linter on it also.